src/tribol/mesh/MfemData.hpp

```diff
@@ -0,0 +1,1390 @@
+   * 
+   * @param src Grid function on (coarse) submesh
+   * @return mfem::GridFunction on (refined) redecomp mesh
+   */
+  mfem::GridFunction SubmeshToRedecomp(const mfem::ParGridFunction& src) const;
+
+  /**
+   * @brief Transfer grid function on redecomp mesh to grid function on submesh
+   * 
+   * @param src Grid function on (refined) redecomp mesh
+   * @return mfem::ParGridFunction on (coarse) submesh
+   */
+  mfem::ParGridFunction RedecompToSubmesh(const mfem::GridFunction& src) const;
+
+  /**
+   * @brief Transfer grid function on redecomp mesh to grid function on submesh
+   * 
+   * @param src Grid function on (refined) redecomp mesh
+   * @param dst Zero-valued grid function on (coarse) submesh
+   */
+  void RedecompToSubmesh(
+    const mfem::GridFunction& src,
+    mfem::ParGridFunction& dst
+  ) const;
+
+  /**
+   * @brief Get the (coarse) Submesh associated with the SubmeshRedecompTransfer object
+   * 
+   * @return const mfem::ParSubMesh& 
+   */
+  const mfem::ParSubMesh& GetSubmesh() const 
+  { 
+    return static_cast<const mfem::ParSubMesh&>(*submesh_fes_.GetParMesh());
+  }
+
+  /**
+   * @brief Set a new (refined) redecomp mesh associated with the existing (refined) submesh
+   * 
+   * @param redecomp RCB redecomposed (refined) submesh
+   */
+  void UpdateRedecomp(redecomp::RedecompMesh& redecomp);
+
+private:
+  /**
+   * @brief Create a finite element space on the (refined) redecomp mesh
+   * 
+   * @param redecomp RCB redecomposed (refined) submesh
+   * @param submesh_fes Original finite element space on the (refined) submesh
+   * @return std::unique_ptr<mfem::FiniteElementSpace> 
+   */
+  static std::unique_ptr<mfem::FiniteElementSpace> CreateRedecompFESpace(
+    redecomp::RedecompMesh& redecomp,
+    mfem::ParFiniteElementSpace& submesh_fes
+  );
+
+  /**
+   * @brief Finite element space on the (coarse) submesh 
+   */
+  mfem::ParFiniteElementSpace& submesh_fes_;
+
+  /**
+   * @brief Finite element space on the (refined) redecomp mesh
+   */
+  mutable std::unique_ptr<mfem::FiniteElementSpace> redecomp_fes_;
+
+  /**
+   * @brief Transfer object between low-order and higher-order grid functions
+   */
+  SubmeshLORTransfer* lor_xfer_;
+
+  /**
+   * @brief Transfer object between (refined) submesh and (refined) redecomp mesh
+   */
+  const redecomp::RedecompTransfer redecomp_xfer_;
+};
+
+/**
+ * @brief Facilitates transferring variables from the original mesh to redecomp
+ * levels
+ *
+ * This class simplifies transferring field variables from/to a grid function on
+ * an mfem::ParMesh to/from a grid function on a redecomp::RedecompMesh.  If
+ * transferring to a low-order grid function on a refined mesh is also required,
+ * this class will perform that transfer as well.
+ */
+class ParentRedecompTransfer
+{
+public:
+  /**
+   * @brief Construct a new ParentRedecompTransfer object
+   *
+   * @param parent_fes Finite element space on the original (coarse) mesh
```

Comment:
I don't think coarse means anything here. Does the parent_fes belong to the original volume mfem parmesh registered through the new API function? For clarity, we have a (parent, possibly HO) volume mfem parmesh, a surface ParSubMesh (possibly HO), LOR counterparts of those (I think, and if HO), then serial redecomp meshes. Having to parse coarse and fine is confusing and means different things in different contexts.

Reply:
Updated